### PR TITLE
fix(cache): scope use cache entries by build id

### PR DESCRIPTION
**Pull request by NathanDrake2406**
## What this changes

Shared `"use cache"` entries now include the vinext build ID in their persistent cache key before the cached function ID and serialized arguments. Environments without a build ID keep the previous key shape.

## Why

Persistent handlers such as Workers KV can outlive a deploy. Before this change, vinext keyed shared `"use cache"` entries as function ID plus arguments, so a new deployment could read a serialized RSC result produced by an older function body when the generated ID and arguments stayed stable.

Next.js includes `buildId` as the first part of the `"use cache"` key material:

- [Next.js `CacheKeyParts` tuple includes `buildId`](https://github.com/vercel/next.js/blob/canary/packages/next/src/server/use-cache/use-cache-wrapper.ts#L105-L107)
- [Next.js comment explaining build ID is needed because action IDs are not unique per implementation](https://github.com/vercel/next.js/blob/canary/packages/next/src/server/use-cache/use-cache-wrapper.ts#L1509-L1513)
- [Next.js runtime constructs cache keys with `buildId`](https://github.com/vercel/next.js/blob/canary/packages/next/src/server/use-cache/use-cache-wrapper.ts#L1682-L1684)
- [Next.js custom cache handler test expects `buildId` as the first serialized key element](https://github.com/vercel/next.js/blob/canary/test/e2e/app-dir/use-cache-custom-handler/use-cache-custom-handler.test.ts#L30-L35)

## Approach

`registerCachedFunction()` now reads `process.env.__VINEXT_BUILD_ID` and uses it to scope shared cache entries as build-specific keys. The read keeps a direct `process.env.__VINEXT_BUILD_ID` reference so Vite can inline the value for Worker bundles, and it only recovers the missing `process` global case.

The key builder encodes custom build IDs before placing them in the colon-delimited key, which avoids delimiter collisions without changing key shape when no build ID exists.

## Validation

- `git diff --check`
- `vp test run tests/shims.test.ts -t 'scopes shared cache entries by build ID'`
- `vp test run tests/shims.test.ts -t '"use cache" runtime'`
- `vp check packages/vinext/src/shims/cache-runtime.ts tests/shims.test.ts`
- `vp test run tests/shims.test.ts`
- `vp run vinext#build`

## Risks / follow-ups

This intentionally invalidates existing shared `"use cache"` entries on the next deploy when a build ID is available. That is the desired Next.js-compatible behavior for code-derived cached RSC streams.

### CODE REVIEW

File: packages/vinext/src/shims/cache-runtime.ts [hunk 1]
```diff
@@ -322,6 +338,7 @@ export function registerCachedFunction<T extends (...args: any[]) => Promise<any
   // oxlint-disable-next-line @typescript-eslint/no-explicit-any
   const cachedFn = async (...args: any[]): Promise<any> => {
     const rsc = await getRscModule();
+    const buildId = getUseCacheBuildId();
```
[the remark this answers is not in the document]
Comment (NathanDrake2406) [reply packages/vinext/src/shims/cache-runtime.ts:343]: Fixed in `548d4945`: `registerCachedFunction()` now resolves the build ID once and closes over it for all cache key construction in that registered wrapper.

File: packages/vinext/src/shims/cache-runtime.ts [hunk 2]
```diff
@@ -91,6 +91,22 @@ type RscModule = {
...
 
+function getUseCacheBuildId(): string | undefined {
+  try {
+    // Keep this direct reference so Vite's define transform can inline it for
+    // Worker bundles where the process global might not exist at runtime.
+    return process.env.__VINEXT_BUILD_ID;
+  } catch (error) {
+    if (error instanceof ReferenceError) return undefined;
+    throw error;
+  }
```
[the remark this answers is not in the document]
Comment (NathanDrake2406) [reply packages/vinext/src/shims/cache-runtime.ts:103]: I kept the try/catch deliberately here. A `typeof process` guard is unsafe for the Worker bundle case: if `process.env.__VINEXT_BUILD_ID` is inlined by Vite but `process` is absent at runtime, the guard can return `undefined` before reaching the inlined build ID expression. The valid hot-path concern is addressed by moving `getUseCacheBuildId()` out of `cachedFn`, so this no longer runs per cached invocation.

**Comment by NathanDrake2406** [comment]
/bigbonk review this PR. Focus on blocking correctness issues and test gaps.

**Comment by NathanDrake2406** [comment]
/bigbonk review this PR. Focus on blocking correctness issues and whether the tests cover the real behavior and meaningful edge cases.

File: packages/vinext/src/shims/cache-runtime.ts [hunk 3]
```diff
@@ -91,6 +91,23 @@ type RscModule = {
...
+function getUseCacheBuildId(): string | undefined {
+  try {
+    // Keep this direct reference so Vite's define transform can inline it for
+    // Worker bundles where the process global might not exist at runtime. A
+    // typeof process guard would return before the inlined build ID is reached.
+    return process.env.__VINEXT_BUILD_ID;
+  } catch (error) {
+    if (error instanceof ReferenceError) return undefined;
+    throw error;
+  }
```
Comment (james-elicx) [inline packages/vinext/src/shims/cache-runtime.ts:103]: the try catch here feels odd. if we're inlining this var, shouldn't it be guaranteed to exist?
